File: src/main/java/yunas/batch/DefaultBatchManager.java
package yunas.batch;


import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import yunas.Batch;

import java.util.HashMap;
import java.util.Map;

import static java.lang.System.exit;

/**
 * Manage Batch Modules.
 *
 */
public class DefaultBatchManager implements BatchManager {

    private static final Logger LOG = LoggerFactory.getLogger(DefaultBatchManager.class);

    private Map<String,Batch> modules = new HashMap<>();

    @Override
    public void add(String name, Batch executable) {
        modules.put(name,executable);
    }

    @Override
    public void run(String[] args) {

        try {
            String name = System.getProperty("yunas.batch");
            if (!modules.containsKey(name)) {
                LOG.error("NOT FOUND module: " + name);
                return;
            }
            modules.get(name).action(args);
        } catch (Exception e) {
             LOG.error(e.getMessage());
             exit(1);
        }
    }

}

File: src/main/kotlin/yunas/batch/DefaultBatchManager.kt
package yunas.batch


import org.slf4j.Logger
import org.slf4j.LoggerFactory
import yunas.Batch

import java.util.HashMap

import java.lang.System.exit

/**
 * Manage Batch Modules.
 *
 */
class DefaultBatchManager : BatchManager {

    private val modules = HashMap<String, Batch>()

    override fun add(name: String, executable: Batch) {
        modules.put(name, executable)
    }

    override fun run(args: Array<String>) {

        try {
            val name = System.getProperty("yunas.batch")
            if (!modules.containsKey(name)) {
                LOG.error("NOT FOUND module: " + name)
                return
            }
            modules[name]!!.action(args)
        } catch (e: Exception) {
            LOG.error(e.message)
            exit(1)
        }

    }

    companion object {

        private val LOG = LoggerFactory.getLogger(DefaultBatchManager::class.java)
    }

}